site/app/controllers/forum/ForumController.php

```diff
@@ -345,7 +345,7 @@ public function publishThread() {
         foreach ($_POST["cat"] as $category_id) {
             $categories_ids[] = (int) $category_id;
         }
-        if (empty($thread_title) || empty($thread_post_content)) {
+        if (strlen($thread_title) === 0 || strlen($thread_post_content) === 0) {
```

Comment:
This change only fixes the 0 issue when someone is making a new thread. If someone is replying to a thread by making a post, then the 0 issue still remains 

<img width="537" alt="image" src="https://github.com/Submitty/Submitty/assets/117249183/7ed50436-eb4c-4701-b6c6-56f9c45dfe08">